packages/plugin-zod/src/index.ts

```diff
@@ -0,0 +1,17 @@
+import SchemaBuilder, { BasePlugin, SchemaTypes, GiraphQLOutputFieldConfig } from '@giraphql/core';
+import { ZodFieldWrapper } from './field-wrapper';
+import './global-types';
+
+export default class ZodPlugin<Types extends SchemaTypes> extends BasePlugin<Types> {
+  name: 'GiraphQLZod' = 'GiraphQLZod';
+
+  constructor(builder: GiraphQLSchemaTypes.SchemaBuilder<Types>) {
+    super(builder, 'GiraphQLZod');
+  }
+
+  wrapOutputField(fieldConfig: GiraphQLOutputFieldConfig<Types>) {
+    return new ZodFieldWrapper(fieldConfig);
```

Comment:
should only wrap fields that have a schema, can check fieldConfig.giraphqlOptions for schema and return null if its missing.